packages/core/admin/admin/src/content-manager/pages/ListView/components/BulkActionButtons/tests/index.test.js

```diff
@@ -155,3 +189,122 @@ describe('BulkActionsBar', () => {
   });
 });
+
+const setupConfirmPublish = () => ({
+  ...render(
+    <ConfirmDialogPublishAll
+      isOpen
+      onConfirm={jest.fn()}
+      onToggleDialog={jest.fn()}
+      isConfirmButtonLoading
+    />,
+    {
+      wrapper({ children }) {
+        const client = new QueryClient({
+          defaultOptions: {
+            queries: {
+              retry: false,
+            },
+          },
+        });
+
+        return (
+          <QueryClientProvider client={client}>
+            <ThemeProvider theme={lightTheme}>
+              <IntlProvider locale="en" messages={{}} defaultLocale="en">
+                <Provider store={store}>
+                  <MemoryRouter>
+                    <Table.Root>{children}</Table.Root>
+                  </MemoryRouter>
+                </Provider>
+              </IntlProvider>
+            </ThemeProvider>
+          </QueryClientProvider>
+        );
+      },
+    }
+  ),
+});
+
+describe('ConfirmDialogPublishAll', () => {
+  beforeAll(() => {
+    server.listen();
+  });
+
+  afterAll(() => {
+    server.close();
+  });
+
+  it('should show a default message if there are not draft relations', async () => {
+    const { findByText, queryByText } = setupConfirmPublish();
+    expect(await findByText('Confirmation')).toBeInTheDocument();
+    expect(
+      queryByText('not published yet and might lead to unexpected behavior')
+    ).not.toBeInTheDocument();
+
+    expect(await findByText('Are you sure you want to publish these entries?')).toBeInTheDocument();
+  });
+
+  it('should show the warning message with just 1 draft relation and two entries', async () => {
+    server.use(
+      rest.get('*/countManyEntriesDraftRelations', (req, res, ctx) => {
+        return res.once(
+          ctx.status(200),
+          ctx.json({
+            data: 1,
+          })
+        );
+      })
+    );
+
+    const { findByText, getByTestId } = setupConfirmPublish();
+    expect(await findByText('Confirmation')).toBeInTheDocument();
+    const draftWarning = await getByTestId('draft-rel-warning');
+
+    expect(draftWarning.textContent).toBe('1  relation  out of 2  entries   is ');
+  });
+
+  it('should show the warning message with 2 draft relations and two entries', async () => {
+    server.use(
+      rest.get('*/countManyEntriesDraftRelations', (req, res, ctx) => {
+        return res.once(
+          ctx.status(200),
+          ctx.json({
+            data: 2,
+          })
+        );
+      })
+    );
+
+    const { findByText, getByTestId } = setupConfirmPublish();
+    expect(await findByText('Confirmation')).toBeInTheDocument();
+    const draftWarning = await getByTestId('draft-rel-warning');
+
+    expect(draftWarning.textContent).toBe('2  relations  out of 2  entries   are ');
+  });
+
+  it('should not show the Confirmation component if there is an error coming from the API', async () => {
+    const originalConsoleError = console.error;
+    console.error = jest.fn();
```

Comment:
How come we're mocking the console?

Reply:
because otherwise I have warning in the console when I run the unit tests

Reply:
Okay, but if the test throws an error, no one will know that this test is failing? I'd be interested to hear what the others think cc @gu-stav / @markkaylor / @remidej / @Feranchz

Reply:
Is the error thrown by axios? In this case I think this is fine and what we do in other places of the codebase. We could start a discussion at some point if there is a way to disable this kind of console-style error reporting from axios for our tests, because it creates the need in different places to overwrite the logging in order to keep the tests clean which I find annoying.

Reply:
At the end it doesn't prevent to log the error in the console so I removed the mock